### DISCUSSION TITLE
Incoming mail log fragmentation? - log too big

### DISCUSSION THREAD

**Discussion by ajtonyb** [postalserver/postal#2544] [Help with using Postal] [answered]
Hey Guys!
Is there any way currently to fragment logs into smaller pieces for example every month? Our current log file is running since the beginning of the year, and it is about 50 GB. Grepping this file took about 2-3 minutes every time. Zipping and moving such a big file every time is also very tough and time consuming.
I looked in the docs and on github but did not find anything regarding this. Did I prehaps miss something?

**Comment by willpower232** [comment]
Which log file is this sorry? What is the path?

This might be a job for logrotate

**Reply by ajtonyb** [reply]
It is:
/var/lib/docker/containers/2af31... this looks like a hash and than there's a json.log within this with the same name.
But yeah, I guess logrotate could take care of it.

**Reply by willpower232** [reply]
that is a docker volume so shouldn't be used outside of the container, what is the end of the path?

**Reply by ajtonyb** [reply]
Sorry, I was out of office for a while, then waited for the logrotate to run.

So, I did the logrotate and it worked good.
But unfortunately, logging stopped after. Had to restart the postal-smtp container, to make it resume logging. Is there a solve for this, or known cause?
Worst case a restart can be scheduled.

**Reply by willpower232** [reply]
> that is a docker volume so shouldn't be used outside of the container, what is the end of the path?

**Reply by ajtonyb** [reply]
It's the log of the container:
2af31ae1a4e198f25...34ab7363ede3783790a5-json.log
If this is what you mean.

**Reply by willpower232** [reply, marked as answer]
Ah right I understand now.

As I said, editing the files without involving docker is probably what broke it but it looks like docker-compose has these options built in.

Try editing your docker-compose file so it looks like this (i.e. add those last three lines to each service listed in the file)

```
  worker:
    image: ghcr.io/postalserver/postal:{{version}}
    command: postal worker
    network_mode: host
    volumes:
      - /opt/postal/config:/config
    logging:
      options:
        max-size: 50m
```

@adamcooke probably worth including this in the install repo as this will be a problem for everyone.

**Reply by ajtonyb** [reply]
> logging:
      options:
        max-size: 50m

In this case what does the 50m stand for, MBs?
Also, I have seen, that other log-drivers are available. Would changing it from json-file to i.e.: local interfere with the operation of postal?

**Reply by ajtonyb** [reply]
Sorry, stupid question, of course it stands for MB

**Reply by willpower232** [reply]
I hope it stands for MB but I guess we'll see, I only found that on stack overflow of course.

I don't think there would be an issue changing it from json-file however in terms of "out of the box" it should probably be the default for ease of use with new users and then advanced users can change where the logging happens.

**Reply by ajtonyb** [reply]
One more thing. Last week, I changed the docker-compose.yml file to hopefully make it logrotate. And I recomposed using docker-compose up. 
From that time, some containers appeared (worker-1, requeuer-1, cron-1) with the beginning "install"(this might be because the docker-compose.yml is in the install folder). I don't know anything else about them. Should I have restarted postal using "postal stop" and "postal start" to apply the changes?
If it's not obvious, I'm still learning docker.

**Reply by willpower232** [reply]
your analysis is correct, the ones named "install" are probably not working that great if the "real" ones are still going and will be fighting over ports.

If you go into the install folder and `docker-compose down` then `postal stop` and `postal start`, that should do the trick.
